### DISCUSSION TITLE
Insert cursor bar style

### DISCUSSION THREAD

**Discussion by ghost** [helix-editor/helix#11481] [General]
Hello, i was wondering if it is possible to style the insert cursor bar? It is extremely thin and has a dark gray color which makes it very very hard for me to spot, ideally i'd like to change its thickness and color, but this does not seem to be possible with the ``ui.cursor.insert`` and ``ui.cursor.primary.insert`` options. Show me da wae

If it isn't currently possible, lets keep this issue open for tracking, i might take a stab at it when i have some time.

**Comment by kirawi** [comment]
The style of the bar cursor is set by your terminal emulator. Programs can't change it.

**Reply by ghost** [reply]
my terminal cursor bar and helix cursor bar is vastly different, are you suggesting this is entirely at the hand of the terminal?

**Reply by kirawi** [reply]
Yes, there is no way to style bar cursors.

**Reply by ghost** [reply]
okay thank you, im using a terminal that's in beta so there is probably some bug then, will have to take it up with them!

**Reply by kirawi** [reply]
I actually want to correct myself here. It is possible on some terminals to change the color. I will investigate why it doesn't.

**Reply by ghost** [reply]
fyi i'm also able to configure the thickness of the cursor and it's blinking style, i use ghostty
